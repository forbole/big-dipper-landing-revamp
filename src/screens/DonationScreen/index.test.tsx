/* eslint-disable @next/next/no-img-element */
/* eslint-disable react/display-name */
import { render, screen } from '@testing-library/react';
import { motion } from 'framer-motion';
import { MockTheme } from '~tests/mocks';
import Component from '.';

const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('~src/assets/copy.svg', () => () => <svg data-testid="copy" />);
jest.mock('~src/components/LayoutComponent', () => (props: object) => (
  <motion.div data-testid="Layout" {...props} />
));
jest.mock('~src/components/SectionBox', () =>
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  ({ main, ...props }: { [p: string]: unknown }) => (
    <motion.div data-testid="SectionBox" {...props} />
  )
);
jest.mock('~src/components/SectionLimit', () => (props: object) => (
  <motion.div data-testid="SectionLimit" {...props} />
));
jest.mock('~src/components/ContentBox', () => (props: object) => (
  <motion.div data-testid="ContentBox" {...props} />
));
// ==================================
// unit tests
// ==================================
describe('Donation', () => {
  it('matches snapshot', async () => {
    const { container } = render(
      <MockTheme>
        <Component />
      </MockTheme>
    );

    expect(screen.getByText(/donation/i)).toBeInTheDocument();
    expect(screen.getByText(/description/i)).toBeInTheDocument();
    expect(screen.getByTestId('Layout')).toBeInTheDocument();
    expect(screen.getByTestId('SectionBox')).toBeInTheDocument();
    expect(screen.getByTestId('SectionLimit')).toBeInTheDocument();
    expect(screen.getAllByTestId('ContentBox').length).toBe(3);
    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
