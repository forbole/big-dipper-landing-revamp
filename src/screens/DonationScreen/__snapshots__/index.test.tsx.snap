// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Donation matches snapshot 1`] = `
.emotion-0 .sectionlimit__container {
  padding: 0;
}

.emotion-0 .donation__title {
  margin-bottom: 32px;
  text-align: center;
}

@media (min-width:768px) {
  .emotion-0 .donation__title {
    text-align: left;
    margin-bottom: 40px;
  }
}

.emotion-0 .donation__description {
  margin-bottom: 32px;
}

.emotion-0 .donation__address-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  gap: 24px;
  text-align: center;
  margin-bottom: 24px;
}

@media (max-width:767.95px) {
  .emotion-0 .donation__address-container {
    -webkit-box-flex-flow: column nowrap;
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
  }
}

.emotion-0 .donation__address-logo {
  margin-bottom: 24px;
}

.emotion-0 .donation__address-key {
  margin-bottom: 8px;
}

.emotion-0 .donation__address-address {
  word-break: break-all;
  margin-bottom: 24px;
}

.emotion-0 .donation__address-button-copy {
  margin-right: 8px;
}

.emotion-0 .donation__address-button {
  padding: 0.75rem 1rem;
  border-radius: 200px;
  background-color: #F6504B;
  background-image: linear-gradient(103.18deg, #F6504B 20.77%, #DC001A 118.69%);
  color: #FFFFFF;
  width: 100%;
}

.emotion-0 .donation__address-button:hover {
  -webkit-filter: brightness(1.2);
  filter: brightness(1.2);
}

.emotion-1 {
  margin: 0;
  font-size: 2rem;
  font-weight: 600;
  font-family: 'SF Pro Text',-apple-system,BlinkMacSystemFont,sans-serif;
  line-height: 1.2;
}

.emotion-2 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  font-family: 'SF Pro Text',-apple-system,BlinkMacSystemFont,sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

.emotion-3 {
  margin: 0;
  font-size: 1.25rem;
  font-weight: 600;
  font-family: 'SF Pro Text',-apple-system,BlinkMacSystemFont,sans-serif;
  line-height: 1.235;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-style: normal;
  font-weight: 600;
  font-size: 0.875rem;
  line-height: 1.214;
  font-family: 'SF Pro Text',-apple-system,BlinkMacSystemFont,sans-serif;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: rgba(0, 0, 0, 0.87);
  background-color: #0939B0;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
}

.emotion-5::-moz-focus-inner {
  border-style: none;
}

.emotion-5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-5 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #202A43;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-5:hover {
    background-color: #0939B0;
  }
}

.emotion-5:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-5.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-5.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
  box-shadow: none;
  background-color: rgba(255, 255, 255, 0.12);
}

.emotion-6 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    data-testid="Layout"
  >
    <div
      class="emotion-0"
      data-testid="SectionBox"
    >
      <div
        data-testid="SectionLimit"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 donation__title emotion-1"
        >
          donation
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 donation__description emotion-2"
        >
          description
        </p>
        <div
          class="donation__address-container"
        >
          <div
            data-testid="ContentBox"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2760%27%20height=%2760%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="btc"
                class="donation__address-logo"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <h4
              class="MuiTypography-root MuiTypography-h4 donation__address-key emotion-3"
            >
              btc
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 donation__address-address emotion-2"
            >
              bc1qye4k27zsn5nehzded6jwsvzg8qd6kgvxyhckts
            </p>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium donation__address-button emotion-5"
              tabindex="0"
              type="button"
            >
              <svg
                data-testid="copy"
              />
              copy
              <span
                class="MuiTouchRipple-root emotion-6"
              />
            </button>
          </div>
          <div
            data-testid="ContentBox"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2760%27%20height=%2760%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="eth"
                class="donation__address-logo"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <h4
              class="MuiTypography-root MuiTypography-h4 donation__address-key emotion-3"
            >
              eth
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 donation__address-address emotion-2"
            >
              0x8CAb9F3fC6bBBD819050365627FC6B79d0ea73e6
            </p>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium donation__address-button emotion-5"
              tabindex="0"
              type="button"
            >
              <svg
                data-testid="copy"
              />
              copy
              <span
                class="MuiTouchRipple-root emotion-6"
              />
            </button>
          </div>
          <div
            data-testid="ContentBox"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2760%27%20height=%2760%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="atom"
                class="donation__address-logo"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <h4
              class="MuiTypography-root MuiTypography-h4 donation__address-key emotion-3"
            >
              atom
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 donation__address-address emotion-2"
            >
              cosmos1n67vdlaejpj3uzswr9qapeg76zlkusj5k875ma
            </p>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium donation__address-button emotion-5"
              tabindex="0"
              type="button"
            >
              <svg
                data-testid="copy"
              />
              copy
              <span
                class="MuiTouchRipple-root emotion-6"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
